### PR TITLE
NOX-557 Custom Domain Events

### PR DESCRIPTION
**Pull request by msadikkoseaupa**

### CODE REVIEW

File: tests/Nox.Generator.Tests/files/yaml/design/api/Entities/country.entity.nox.yaml [hunk 1]
```diff
@@ -129,4 +129,8 @@ persistence:
         raiseEvents: domainAndIntegrationEvents
     delete:
         isEnabled: true
-        raiseEvents: domainAndIntegrationEvents
\ No newline at end of file
+        raiseEvents: domainAndIntegrationEvents
+        
+events:
+   - name: TestEvent
```
Comment (gjorgji-grgovski-authoritypartners) [inline tests/Nox.Generator.Tests/files/yaml/design/api/Entities/country.entity.nox.yaml]: Why this event is not generated? Please use some better names for the examples.

File: src/Nox.Generator/Domain/DomainEventGenerator/DomainEventGeneratorBack.cs
Comment (gjorgji-grgovski-authoritypartners) [inline src/Nox.Generator/Domain/DomainEventGenerator/DomainEventGeneratorBack.cs:1]: This is a leftover I guess the old generator? Please remove it

**Review by gjorgji-grgovski-authoritypartners — CHANGES_REQUESTED** [review]

**Comment by gjorgji-grgovski-authoritypartners** [comment]
Don't forget to generate documentation for the custom domain events

File: src/Nox.Solution/Models/Domain/Entities/Events/DomainEvent.cs [hunk 2]
```diff
@@ -4,9 +4,15 @@
...
 [GenerateJsonSchema]
-public class DomainEvent: NoxComplexTypeDefinition
+public class DomainEvent
 {
-    [Title("Whether or not to also raise this as an application event.")]
-    [Description("Will raise this event both as an integration and application event")]
-    public bool RaiseApplicationEvent { get; set; } = false;
+    [Required]
+    [Title("Event name")]
+    [Description("Assign a descriptive name to the event. Should be a singular noun and be unique within a collection of events. PascalCase recommended.")]
```
Comment (rochar) [inline src/Nox.Solution/Models/Domain/Entities/Events/DomainEvent.cs]:
```suggestion
    [Description("Descriptive name for the domain event. Should be a singular noun and be unique within a collection of events. Pascal Case recommended.")]
```

**Review by rochar — CHANGES_REQUESTED** [review]

File: tests/Nox.Generator.Tests/files/yaml/design/api/Entities/country.entity.nox.yaml (hunk 1 above)
Comment (msadikkoseaupa) [reply tests/Nox.Generator.Tests/files/yaml/design/api/Entities/country.entity.nox.yaml]: This test does not generate file, and this lines have been removed.

File: src/Nox.Generator/Domain/DomainEventGenerator/DomainEventGeneratorBack.cs
Comment (msadikkoseaupa) [reply src/Nox.Generator/Domain/DomainEventGenerator/DomainEventGeneratorBack.cs:1]: Desired changes have been applied

**Review by rochar — DISMISSED** [review]

**Review by gjorgji-grgovski-authoritypartners — DISMISSED** [review]

**Review by rochar — DISMISSED** [review]

**Review by gjorgji-grgovski-authoritypartners — DISMISSED** [review]

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

**Review by rochar — APPROVED** [review]

**Pull request merged**